### PR TITLE
[TOOLS-690] ScriptUtils / AssigneeAndWipValidator / ParentValidator / doTransition / Error messages improvement / WorkflowScripts

### PR DESCRIPTION
**Pull request by jhonypg**

### CODE REVIEW

File: src/main/java/br/objective/jira/workflow/validators/IssueTransitionValidator.java [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 import static br.objective.jira.workflow.utils.TransitionUtils.getTransitionActionDescriptor;
 import static br.objective.jira.workflow.utils.TransitionUtils.isValidTransition;
+import static org.apache.commons.lang.StringUtils.isBlank;
```
Comment (herbert-venancio) [inline src/main/java/br/objective/jira/workflow/validators/IssueTransitionValidator.java]: If possible, change to org.apache.commons.lang**3**.StringUtils.isBlank

File: src/main/java/br/objective/jira/workflow/validators/ValidatorsFacade.java [hunk 2]
```diff
@@ -10,11 +10,11 @@
 
 public class ValidatorsFacade {
 
-	public static void parentValidation(Issue childIssue, String transitionName) throws Throwable {
-		if (childIssue == null || childIssue.getParentObject() == null)
+	public static void parentValidation(Issue subtask, String transitionName) throws Throwable {
+		if (subtask == null || subtask.getParentObject() == null)
 			throw new IllegalArgumentException("Error on parentValidation: the \"childIssue\" and the \"parentIssue\" cannot be null.");
```
Comment (herbert-venancio) [inline src/main/java/br/objective/jira/workflow/validators/ValidatorsFacade.java]: \\"childIssue\\" was intentional?

File: src/main/java/br/objective/jira/workflow/conditions/IssuesCondition.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.atlassian.jira.issue.Issue;
+
+class IssuesCondition {
+
+	public static boolean isIssuesOnStatusCategory(Collection<Issue> issues, String statusCategory, String... issueTypeIdsToFilter) {
+		Collection<Issue> issuesToSearch;
```
[the remark this answers is not in the document]
Comment (jhonypg) [reply src/main/java/br/objective/jira/workflow/conditions/IssuesCondition.java]: Ok

File: src/main/java/br/objective/jira/workflow/validators/IssueTransitionValidator.java (hunk 1 above)
Comment (jhonypg) [reply src/main/java/br/objective/jira/workflow/validators/IssueTransitionValidator.java]: Ok

File: src/main/java/br/objective/jira/workflow/postFunctions/PostFunctionsFacade.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+import br.objective.jira.rest.LoggedUser;
+import br.objective.jira.workflow.utils.TransitionUtils;
+
+public class PostFunctionsFacade {
+
+	public static void doTransitionOnParent(Issue subtask, String transitionName) throws IllegalArgumentException {
+		if (subtask == null || subtask.getParentObject() == null)
+			throw new IllegalArgumentException("Error on parentValidation: the \"childIssue\" and the \"parentIssue\" cannot be null.");
+		TransitionUtils.doTransition(subtask.getParentObject(), transitionName, LoggedUser.get());
```
[the remark this answers is not in the document]
Comment (jhonypg) [reply src/main/java/br/objective/jira/workflow/postFunctions/PostFunctionsFacade.java]: Ok

File: src/main/java/br/objective/jira/workflow/validators/ValidatorsFacade.java (hunk 2 above)
Comment (jhonypg) [reply src/main/java/br/objective/jira/workflow/validators/ValidatorsFacade.java]: Changed to "subtask".

**Pull request merged**